packages/core/src/components/Footer/Footer.styles.tsx

```diff
@@ -1,74 +1,51 @@
-import styled from "@emotion/styled";
-
-import { Breakpoints as MuiBreakpoints } from "@mui/material";
-
 import { theme } from "@hitachivantara/uikit-styles";
-
-import { HvTypography } from "@core/components/Typography";
-import { transientOptions } from "@core/utils/transientOptions";
-
-export const StyledRoot = styled(
-  "footer",
-  transientOptions
-)(({ $breakpoints }: { $breakpoints: MuiBreakpoints }) => ({
-  display: "flex",
-  alignItems: "center",
-  flexDirection: "row",
-  padding: `${theme.space.xs} ${theme.space.sm}`,
-  height: 40,
-  bottom: 0,
-  zIndex: theme.zIndices.base,
-  backgroundColor: theme.colors.atmo1,
-  boxShadow: `0 -1px 0 ${theme.colors.atmo4}`,
-  [$breakpoints.down("sm")]: {
-    height: "unset",
-    flexDirection: "column",
-    padding: theme.space.xs,
+import { createClasses } from "@core/utils/classes";
+
+export const { staticClasses, useClasses } = createClasses("HvFooter", {
+  root: {
+    display: "flex",
+    alignItems: "center",
+    flexDirection: "row",
+    padding: `${theme.space.xs} ${theme.space.sm}`,
+    height: 40,
+    bottom: 0,
+    zIndex: theme.zIndices.base,
+    backgroundColor: theme.colors.atmo1,
+    boxShadow: `0 -1px 0 ${theme.colors.atmo4}`,
+    "&$small": {
+      height: "unset",
+      flexDirection: "column",
+      padding: theme.space.xs,
+    },
   },
-}));
-
-export const StyledName = styled(
-  HvTypography,
-  transientOptions
-)(({ $breakpoints }: { $breakpoints: MuiBreakpoints }) => ({
-  [$breakpoints.down("sm")]: {
-    marginBottom: theme.space.xs,
+  name: {
+    "&$small": {
+      marginBottom: theme.space.xs,
+    },
   },
-}));
-
-export const StyledRightContainer = styled(
-  "div",
-  transientOptions
-)(({ $breakpoints }: { $breakpoints: MuiBreakpoints }) => ({
-  display: "flex",
-  alignItems: "center",
-  flexDirection: "row",
-  marginLeft: "auto",
-  [$breakpoints.down("sm")]: {
-    flexDirection: "column",
-    textAlign: "center",
-    marginLeft: "unset",
+  rightContainer: {
+    display: "flex",
+    alignItems: "center",
+    flexDirection: "row",
+    marginLeft: "auto",
+    "&$small": {
+      flexDirection: "column",
+      textAlign: "center",
+      marginLeft: "unset",
+    },
   },
-}));
-
-export const StyledCopyright = styled(
-  HvTypography,
-  transientOptions
-)(({ $breakpoints }: { $breakpoints: MuiBreakpoints }) => ({
-  [$breakpoints.down("sm")]: {
-    flexDirection: "column",
-    marginBottom: theme.space.xs,
+  copyright: {
+    "&$small": {
+      flexDirection: "column",
+      marginBottom: theme.space.xs,
+    },
   },
-}));
-
-export const StyledSeparator = styled("div")(
-  ({ $breakpoints }: { $breakpoints: MuiBreakpoints }) => ({
+  separator: {
     width: 1,
     height: 16,
     backgroundColor: theme.colors.secondary,
     margin: `${theme.space.xs} ${theme.space.sm}`,
-    [$breakpoints.down("sm")]: {
-      display: "none",
-    },
-  })
-);
+    "&$small": { display: "none" },
+  },
+  small: {},
```

Comment:
Regarding the point below about `classes.small` being in a single element, it would also simply styling here:
```suggestion
  small: {
    "&$root": {
      height: "unset",
      flexDirection: "column",
      padding: theme.space.xs,
    },
    "&$name": ...
  },
```

([example on another conditional class](https://github.com/lumada-design/hv-uikit-react/blob/master/packages/core/src/components/BaseCheckBox/BaseCheckBox.styles.ts#L15-L26))